metaflow/runtime.py

```diff
@@ -238,8 +245,32 @@ def execute(self):
                                         'by the end of flow.')
 
     def _killall(self):
+        # If we are here, all children have received a signal and are shutting down.
+        # We want to give them an opportunity to do so and then kill
+        live_workers = []
+        temp_workers = {}
         for worker in self._workers.values():
-            worker.kill()
+            temp_workers[worker] = 1
+        live_workers = temp_workers.keys()
+        now = int(time.time())
+        self._logger('Waiting for %d workers to shutdown...' % len(live_workers),
```

Comment:
The notion of workers is not exposed to users of Metaflow. Maybe reword this or take this out entirely?

Reply:
Similar for L:266 and L:272.

Reply:
I can maybe say something like "Terminating %d running steps...". I do want to give some feedback to make sure the user feels something is happening.